llvm/test/Instrumentation/AddressSanitizer/AMDGPU/asan-static-lds-test.ll

```diff
@@ -0,0 +1,249 @@
+; CHECK:       [[ASAN_REPORT1]]:
+; CHECK-NEXT:    br i1 [[TMP70]], label %[[BB73:.*]], label %[[BB74:.*]]
+; CHECK:       [[BB73]]:
+; CHECK-NEXT:    call void @__asan_report_store_n(i64 [[TMP61]], i64 4) #[[ATTR7]]
+; CHECK-NEXT:    call void @llvm.amdgcn.unreachable()
+; CHECK-NEXT:    br label %[[BB74]]
+; CHECK:       [[BB74]]:
+; CHECK-NEXT:    br label %[[BB75]]
+; CHECK:       [[BB75]]:
+; CHECK-NEXT:    [[TMP76:%.*]] = ptrtoint ptr addrspace(3) [[TMP57]] to i32
+; CHECK-NEXT:    [[TMP77:%.*]] = load ptr addrspace(1), ptr addrspace(3) @llvm.amdgcn.sw.lds.k0, align 8
+; CHECK-NEXT:    [[TMP78:%.*]] = getelementptr inbounds i8, ptr addrspace(1) [[TMP77]], i32 [[TMP76]]
+; CHECK-NEXT:    [[TMP79:%.*]] = ptrtoint ptr addrspace(1) [[TMP78]] to i64
+; CHECK-NEXT:    [[TMP80:%.*]] = lshr i64 [[TMP79]], 3
+; CHECK-NEXT:    [[TMP81:%.*]] = add i64 [[TMP80]], 2147450880
+; CHECK-NEXT:    [[TMP82:%.*]] = inttoptr i64 [[TMP81]] to ptr
+; CHECK-NEXT:    [[TMP83:%.*]] = load i8, ptr [[TMP82]], align 1
+; CHECK-NEXT:    [[TMP84:%.*]] = icmp ne i8 [[TMP83]], 0
+; CHECK-NEXT:    [[TMP85:%.*]] = and i64 [[TMP79]], 7
+; CHECK-NEXT:    [[TMP86:%.*]] = trunc i64 [[TMP85]] to i8
+; CHECK-NEXT:    [[TMP87:%.*]] = icmp sge i8 [[TMP86]], [[TMP83]]
+; CHECK-NEXT:    [[TMP88:%.*]] = and i1 [[TMP84]], [[TMP87]]
+; CHECK-NEXT:    [[TMP89:%.*]] = call i64 @llvm.amdgcn.ballot.i64(i1 [[TMP88]])
+; CHECK-NEXT:    [[TMP90:%.*]] = icmp ne i64 [[TMP89]], 0
+; CHECK-NEXT:    br i1 [[TMP90]], label %[[ASAN_REPORT2:.*]], label %[[BB93:.*]], !prof [[PROF1]]
+; CHECK:       [[ASAN_REPORT2]]:
+; CHECK-NEXT:    br i1 [[TMP88]], label %[[BB91:.*]], label %[[BB92:.*]]
+; CHECK:       [[BB91]]:
+; CHECK-NEXT:    call void @__asan_report_store_n(i64 [[TMP79]], i64 4) #[[ATTR7]]
+; CHECK-NEXT:    call void @llvm.amdgcn.unreachable()
+; CHECK-NEXT:    br label %[[BB92]]
+; CHECK:       [[BB92]]:
+; CHECK-NEXT:    br label %[[BB93]]
+; CHECK:       [[BB93]]:
+; CHECK-NEXT:    store i32 8, ptr addrspace(3) [[TMP36]], align 2
+; CHECK-NEXT:    br label %[[CONDFREE:.*]]
+; CHECK:       [[CONDFREE]]:
+; CHECK-NEXT:    call void @llvm.amdgcn.s.barrier()
+; CHECK-NEXT:    br i1 [[XYZCOND]], label %[[FREE:.*]], label %[[END:.*]]
+; CHECK:       [[FREE]]:
+; CHECK-NEXT:    [[TMP94:%.*]] = load ptr addrspace(1), ptr addrspace(3) @llvm.amdgcn.sw.lds.k0, align 8
+; CHECK-NEXT:    [[TMP95:%.*]] = call ptr @llvm.returnaddress(i32 0)
+; CHECK-NEXT:    [[TMP96:%.*]] = ptrtoint ptr [[TMP95]] to i64
+; CHECK-NEXT:    [[TMP97:%.*]] = ptrtoint ptr addrspace(1) [[TMP94]] to i64
+; CHECK-NEXT:    call void @__asan_free_impl(i64 [[TMP97]], i64 [[TMP96]])
+; CHECK-NEXT:    br label %[[END]]
+; CHECK:       [[END]]:
+; CHECK-NEXT:    ret void
+;
+WId:
+  %0 = call i32 @llvm.amdgcn.workitem.id.x()
+  %1 = call i32 @llvm.amdgcn.workitem.id.y()
+  %2 = call i32 @llvm.amdgcn.workitem.id.z()
+  %3 = or i32 %0, %1
+  %4 = or i32 %3, %2
+  %5 = icmp eq i32 %4, 0
+  br i1 %5, label %Malloc, label %14
+
```

Comment:
There's a lot going on in these test functions. Can you use simple tests that show single operations? Also need to test an arbitrary call user of a pointer